src/main/java/teammates/ui/webapi/GetFeedbackSessionAction.java

```diff
@@ -71,4 +70,33 @@ public JsonResult execute() {
 
         return new JsonResult(response);
     }
+
+    private FeedbackSessionAttributes getFeedbackSessionAttributesWithCorrectDeadlines(Intent intent) {
+        String courseId = getNonNullRequestParamValue(Const.ParamsNames.COURSE_ID);
+        String feedbackSessionName = getNonNullRequestParamValue(Const.ParamsNames.FEEDBACK_SESSION_NAME);
+        FeedbackSessionAttributes feedbackSessionAttributes = getNonNullFeedbackSession(feedbackSessionName, courseId);
+
+        switch (intent) {
+        case STUDENT_SUBMISSION:
```

Comment:
I have tried it actually, you can see it in my ["Combine similar methods that check feedback session state"](https://github.com/TEAMMATES/teammates/pull/11606/commits/b4530946bfe014098775a5fc3613b230621f19f6) commits (though it's kind of outdated now). It requires quite a bit of code repetition which makes it pretty ugly, so I opted to just abstract it away. If the `INSTRUCTOR_SUBMISSION` case used `response.hideInformationForInstructor()` instead of `response.hideInformationForStudent()`, I think merging the two methods would be better, but it unfortunately does not. I can still merge the two methods though if you still believe they are better merged.

Reply:
For clarity, here is how it would look like if the methods were merged in its current state using one switch case block. It is pretty ugly in my opinion. Though, maybe I am going about this wrongly.

![image](https://user-images.githubusercontent.com/65202977/156938046-823a3e69-bdc0-47e7-8a06-6f31d025113b.png)

An alternative way to do it would be with two switch cases as follows. However, it is not that different from just abstracting it away I feel.

![image](https://user-images.githubusercontent.com/65202977/156938200-5e082be4-fcda-49b9-bea4-1d6b5d72fe89.png)

Still, if you prefer either of these, I can commit it.